neural_compressor/torch/algorithms/weight_only/autoround.py

```diff
@@ -12,27 +12,141 @@
+            4. parameter name changed, such as 'packed_weight' -> 'qweight'.
+            5. zeros is always needed even for sym.
+        inplace (bool, optional): Compress the model in place, or copy the model and compress it.
+    """
+    if inplace:
+        compressed_model = model
+    else:
+        compressed_model = copy.deepcopy(model)
+    if isinstance(weight_config, str):
+        with open(weight_config, "r") as f:
+            q_config = json.load(f)
+    else:
+        q_config = weight_config
+    for k, v in q_config.items():
+        logger.info(f"Packing {k}")
+        if "float" in v["data_type"]:
+            continue
+        dtype = v["data_type"]
+        num_bits = v["bits"]
+        group_size = v["group_size"]
+        sym = v["sym"]
+        scale_dtype = v["scale_dtype"]
+        m = get_module(compressed_model, k)
+        fp_weight = m.weight.data
+        scale, zp = v["scale"], v["zp"]
+        convert_dtype = scale_dtype
+        if not isinstance(scale, torch.Tensor):
+            scale = torch.tensor(scale, dtype=convert_dtype)
+            zp = torch.tensor(zp, dtype=torch.int32)
+        else:
+            if not inplace:
+                scale = scale.clone()
+                zp = zp.clone()
+                scale = scale.to(dtype=convert_dtype)
+                zp = zp.to(dtype=torch.int32)
+        if is_transformers_imported() and isinstance(m, transformers.Conv1D):
+            fp_weight = fp_weight.t_().contiguous()
+        int_weight = quant_weight_w_scale(fp_weight, scale, zp, group_size, fp_weight.device)
+        int_weight = int_weight.type(torch.int32)
+        if isinstance(m, torch.nn.Linear):
+            in_features = m.in_features
+            out_features = m.out_features
+        elif is_transformers_imported() and isinstance(m, transformers.Conv1D):
+            in_features = m.weight.shape[0]
+            out_features = m.weight.shape[1]
+
+        new_module = WeightOnlyLinear(
+            in_features,
+            out_features,
+            num_bits,
+            group_size,
+            dtype=dtype,
+            scale_dtype=scale_dtype,
+            zp=zp is not None,
+            bias=m.bias is not None,
+            device=device,
+            compression_dtype=compression_dtype,
+            compression_dim=compression_dim,
+            use_optimum_format=use_optimum_format,
+        )
+        new_module.pack(int_weight, scale, zp, m.bias)
+        set_module(compressed_model, k, new_module)
+    return compressed_model
+
+
+class InputCaptureModule(torch.nn.Module):
+
+    def __init__(self, model) -> None:
+        super().__init__()
+        self.data_pairs = []
+        self.device = "cpu"
+        self.orig_model = model
+
+    def forward(self, *args, **kwargs):
+        if args:
+            for arg in args:
+                self.data_pairs.append(arg)
+        if kwargs:
+            self.data_pairs.append(kwargs)
```

Comment:
There might be some potential issues. If the inputs are `input_ids` and `attention_mask`, The `data_pairs` will result in a sequence like [`input_ids`, `attention_mask`, `input_ids`, `attention_mask`, ...].

Reply:
updated, pls review new usage. https://github.com/intel/neural-compressor/pull/1810/files#diff-c9033e8055d324731fffc9533291b83b1b0e3e5f0969deb602dceeb2845a919e